src/messaging/messaging.ts

```diff
@@ -228,6 +237,63 @@ export class Messaging implements MessagingInterface {
     return this.appInternal;
   }
 
+
+  /**
+   * Converts from the public {@link messaging.Message `Message`} type to the
+   * internal/generated {@link protos.google.firebase.fcm.v1.Message `clientMessage`}
+   * type.
+   *
+   * @param message The {@link messaging.Message `Message`} to be converted
+   * @returns A {@link protos.google.firebase.fcm.v1.Message `clientMessage`},
+   * where like fields are the same as the inptuted message.
+   */
+  private convertToClientMessage(message: Message): ClientMessage {
+    //TODO: Add conversion for android, webpush, apns
+
+    const clientMessage: ClientMessage = {
+      data: message.data,
+      notification: message.notification,
+      fcmOptions: message.fcmOptions
+    }
+
+    if ('token' in message) {
+      clientMessage.token = message.token;
+    } else if ('topic' in message) {
+      clientMessage.topic = message.topic;
+    } else if ('condition' in message) {
+      clientMessage.condition = message.condition;
+    }
+
+    return clientMessage;
+  }
+
+  /**
+   * Method that returns a fully instantiated service client.
+   *
+   * @returns an instantiated service client with either the application's
+   * service account credentials or the default credentials
+   */
+  private getFcmServiceClient(): FcmServiceClient {
+    if (this.fcmServiceClient) {
+      this.fcmServiceClient;
+    }
+
+    const credential = this.app.options.credential;
+    const options: { credentials?: object; fallback: 'rest' } = {
+      fallback: 'rest'
+    };
+
+    if (credential instanceof ServiceAccountCredential) {
```

Comment:
Pass project ID here instead of as the `parent` field.

Reply:
The same issue as [this comment](https://github.com/firebase/firebase-admin-node/pull/1354#discussion_r679461191) applies here, unfortunately.